assembl/tasks/common.py

```diff
@@ -311,27 +337,334 @@ def create_venv(c, path=None):
+    c.run('mkdir -p %s' % normpath(join(c.config.projectpath, 'var', 'share')))
+    c.run('mkdir -p %s' % get_upload_dir(c))
+
+
+def get_upload_dir(c, path=None):
+    path = path or c.config.get('upload_root', 'var/uploads')
+    if path != '/':
+        path = join(c.config.projectpath, path)
+    return path
+
+
+def chgrp_rec(c, path, group, upto=None):
+    parts = path.split("/")
+    success = False
+    for i in range(len(parts), 1, -1):
+        path = "/".join(parts[:i])
+        if path == upto:
+            break
+        if not c.run('chgrp {group} {path}'.format(group=group, path=path), warn=True).ok:
+            break
+        if not c.run('chmod g+x {path}'.format(path=path), warn=True).ok:
+            break
+        success = True
+    assert success  # At least the full path
+
+
+def filter_autostart_processes(c, processes):
+    return [p for p in processes
+            if c.config.supervisor.get('autostart_' + p, False)]
+
+
+def get_db_dump_name():
+    return 'assembl-backup.pgdump'
+
+
+def remote_db_path(c):
+    return join(c.config.projectpath, get_db_dump_name())
+
+
+def supervisor_process_stop(c, process_name):
+    """
+    Assuming the supervisord process is running, stop one of its processes
+    """
+    print('Asking supervisor to stop %s' % process_name)
+    supervisor_pid_regex = re.compile(r'^\d+')
+    status_regex = re.compile(r'^%s\s*(\S*)' % process_name)
+    with venv(c):
+        supervisord_cmd_result = c.run("supervisorctl pid", warn=True, hide='out')
+    match = supervisor_pid_regex.match(supervisord_cmd_result)
+    if not match:
+        print('Supervisord doesn\'t seem to be running, nothing to stop')
+        return
+    for try_num in range(20):
+        with venv(c):
+            c.run("supervisorctl stop %s" % process_name)
+            status_cmd_result = c.run("supervisorctl status %s" % process_name, hide='out')
+
+        match = status_regex.match(status_cmd_result)
+        if match:
+            status = match.group(1)
+            if(status == 'STOPPED'):
+                print("%s is stopped" % process_name)
+                break
+            if(status == 'FATAL'):
+                print("%s had a fatal error" % process_name)
+                break
+            elif(status == 'RUNNING'):
+                with venv(c):
+                    c.run("supervisorctl stop %s" % process_name)
+            elif(status == 'STOPPING'):
+                print(status)
+            else:
+                print("unexpected status: %s" % status)
+            sleep(1)
+        else:
+            print('Unable to parse status (bad regex?)')
+            print(status_cmd_result)
+            exit()
+
+
+def is_supervisord_running(c):
+    with venv(c):
+        result = c.run('supervisorctl pid')
+    if 'no such file' in result:
+        return False
+    try:
+        pid = int(result)
+        if pid:
+            return True
+    except:
+        return False
+
+
+def webservers_start(c):
```

Comment:
Webserver start/restart are also already in place in deploy.py, can be moved here.